internal/pkg/instrumentation/manager.go

```diff
@@ -97,59 +102,52 @@ func (m *Manager) validateProbeDependents(id probe.ID, symbols []probe.FunctionS
 				return fmt.Errorf("library %s has declared a dependent function %s for probe %s which does not exist, aborting", id, d, s.Symbol)
 			}
 		}
-	}
 
-	return nil
-}
-
-func (m *Manager) registerProbe(p probe.Probe) error {
-	id := p.Manifest().Id
-	if _, exists := m.probes[id]; exists {
-		return fmt.Errorf("library %s registered twice, aborting", id)
-	}
-
-	if err := m.validateProbeDependents(id, p.Manifest().Symbols); err != nil {
-		return err
+		// if no dependency issues, add the symbol to the manager's relevant funcs
+		m.relevantFuncs[s.Symbol] = nil
 	}
 
-	m.probes[id] = p
 	return nil
 }
 
 // GetRelevantFuncs returns the instrumented functions for all managed probes.
 func (m *Manager) GetRelevantFuncs() map[string]interface{} {
-	funcsMap := make(map[string]interface{})
-	for _, i := range m.probes {
-		for _, s := range i.Manifest().Symbols {
-			funcsMap[s.Symbol] = nil
-		}
-	}
-
-	return funcsMap
+	return m.relevantFuncs
 }
 
-// FilterUnusedProbes filterers probes whose functions are already instrumented
-// out of the Manager.
-func (m *Manager) FilterUnusedProbes(target *process.TargetDetails) {
+// FilterUnusedProbesForTarget filters probes whose functions are not instrumented
+// out of the Manager, and updates instrumented probes with Target Details.
+func (m *Manager) FilterUnusedProbesForTarget(target *process.TargetDetails) {
 	existingFuncMap := make(map[string]interface{})
 	for _, f := range target.Functions {
 		existingFuncMap[f.Name] = nil
 	}
 
 	for name, inst := range m.probes {
-		funcsFound := false
-		for _, s := range inst.Manifest().Symbols {
-			if len(s.DependsOn) == 0 {
-				if _, exists := existingFuncMap[s.Symbol]; exists {
-					funcsFound = true
-					break
+		switch p := inst.(type) {
+		case probe.GoLibraryTelemetryProbe:
+			// Filter Probe if unused in target process
+			funcsFound := false
+			for _, s := range p.Manifest().Symbols {
+				if len(s.DependsOn) == 0 {
+					if _, exists := existingFuncMap[s.Symbol]; exists {
+						funcsFound = true
+						break
+					}
 				}
 			}
-		}
 
-		if !funcsFound {
-			m.logger.Debug("no functions found for probe, removing", "name", name)
-			delete(m.probes, name)
+			if !funcsFound {
+				m.logger.Debug("no functions found for probe, removing", "name", name)
+				delete(m.probes, name)
+				continue
+			}
+
+			// If Probe is used, pass target details to Probe
+			p.TargetConfig().TargetDetails = target
```

Comment:
This is setting a value on a returned type from `TargetConfig`. This is not idiomatic of Go, parameters should be passed to function calls not set on returned values.

Reply:
fair enough, my point here was just to pass `TargetDetails` somehow in the config for the probe rather than as a one-time argument, since the target is available at startup and we re-read it a couple times. But, this does codify the 1:1 probe:target ratio so maybe not a good idea